src/main/java/org/apache/maven/model/jdom/JDomDependency.java

```diff
@@ -25,52 +25,86 @@
 
 import java.util.List;
 
+import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_ARTIFACT_ID;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_CLASSIFIER;
+import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_GROUP_ID;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_OPTIONAL;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_SCOPE;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_SYSTEM_PATH;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_TYPE;
+import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_VERSION;
 import static org.apache.maven.model.jdom.util.JDomUtils.getChildElementTextTrim;
 import static org.apache.maven.model.jdom.util.JDomUtils.rewriteElement;
+import static org.codehaus.plexus.util.StringUtils.defaultString;
+import static org.codehaus.plexus.util.StringUtils.trim;
 
 /**
- * JDom implementation of poms DEPENDENCY element
+ * JDOM implementation of the {@link Dependency} class. It holds the child elements of the Maven POMs {@code dependency}
+ * element.
  *
  * @author Robert Scholte (for <a href="https://github.com/apache/maven-release/">Maven Release projct</a>, version 3.0)
+ * @author Marc Rohlfs, CoreMedia AG
  */
-public class JDomDependency extends Dependency implements JDomBacked, MavenCoordinate {
+public class JDomDependency extends Dependency implements JDomBacked {
```

Comment:
Why not adding an `equals` Method here and use it [here](https://github.com/CoreMedia/maven-jdom-parser/blob/b4ef7016e4380c2771f725d57b1026a0c16a0b07/src/main/java/org/apache/maven/model/jdom/JDomDependencies.java#L72)

Reply:
Not sure, but I think this can lead to problems, because we must hold the dependencies list and the regarding JDOM elements synchronous. Problem is that in the POMs the same dependency can actually be defined twice, and we won't notice when the JDK list implementation removes one due to the `equals` implementation.